src/solver/optim/ortoolsImpl/linearProblem.cpp

```diff
@@ -0,0 +1,141 @@
+#include <antares/solver/utils/ortools_utils.h>
+
+namespace Antares::Solver::Optim::OrtoolsImpl
+{
+
+OrtoolsLinearProblem::OrtoolsLinearProblem(bool isMip, const std::string& solverName)
+{
+    auto* mpSolver = isMip
+                  ? MPSolver::CreateSolver((OrtoolsUtils::solverMap.at(solverName)).MIPSolverName)
+                  : MPSolver::CreateSolver((OrtoolsUtils::solverMap.at(solverName)).LPSolverName);
+
+    mpSolver_ = std::unique_ptr<operations_research::MPSolver>(mpSolver);
+}
+
+Api::MipVariable* OrtoolsLinearProblem::addNumVariable(double lb, double ub, const std::string& name)
+{
+    auto* mpVar = mpSolver_->MakeNumVar(lb, ub, name);
+    auto mipVar = std::make_shared<OrtoolsMipVariable>(mpVar);
+
+    if (!mpVar || !mipVar)
+    {
+        logs.error() << "Couldn't add variable to Ortools MPSolver: " << name;
+    }
+
+    if (!variables_.try_emplace(name, mipVar).second)
+    {
+        logs.error() << "Error adding variable: " << name;
+    }
+
+    return mipVar.get();
+}
+
+Api::MipVariable* OrtoolsLinearProblem::addIntVariable(double lb, double ub, const std::string& name)
+{
+    auto* mpVar = mpSolver_->MakeIntVar(lb, ub, name);
+    auto mipVar = std::make_shared<OrtoolsMipVariable>(mpVar);
+
+    if (!mpVar || !mipVar)
+    {
+        logs.error() << "Couldn't add variable to Ortools MPSolver: " << name;
+    }
+
+    if (!variables_.try_emplace(name, mipVar).second)
+    {
+        logs.error() << "Error adding variable: " << name;
+    }
+
+    return mipVar.get();
+}
+
+Api::MipVariable* OrtoolsLinearProblem::getVariable(const std::string& name)
+{
+    return variables_.at(name).get();
+}
+
+Api::MipConstraint* OrtoolsLinearProblem::addConstraint(double lb,
+                                                       double ub,
+                                                       const std::string& name)
+{
+    auto* mpConstraint = mpSolver_->MakeRowConstraint(lb, ub, name);
+    auto mipConstraint = std::make_shared<OrtoolsMipConstraint>(mpConstraint);
+
+    if (!constraints_.try_emplace(name, mipConstraint).second)
+    {
+        logs.error() << "Error adding constraint: " << name;
+    }
+
+    return mipConstraint.get();
+}
+
+Api::MipConstraint* OrtoolsLinearProblem::getConstraint(const std::string& name)
+{
+    return constraints_.at(name).get();
+}
+
+void OrtoolsLinearProblem::setObjectiveCoefficient(Api::MipVariable* var, double coefficient)
+{
+    objective_->setCoefficient(var, coefficient);
+}
+
+double OrtoolsLinearProblem::getObjectiveCoefficient(Api::MipVariable* var)
+{
+    return objective_->getCoefficient(var);
+}
+
+void OrtoolsLinearProblem::setMinimization()
+{
+    objective_->setMinimization();
+}
+
+void OrtoolsLinearProblem::setMaximization()
+{
+    objective_->setMaximization();
+}
+
+Api::MipSolution* OrtoolsLinearProblem::solve()
+{
+    mpSolver_->EnableOutput();
+    auto status = mpSolver_->Solve(*param_);
+
+    std::map<Api::MipVariable*, double> solution;
```

Comment:
Let's be careful to not use pointers as the key : they will not ensure a stable sort order from one execution to the other.

Reply:
Using a map with std::string as key will need MipSolution to have another container for MipVariable ptr
Function like `getOptimalValue` take a MipVariable as arg